### ISSUE TITLE
aria-required missing for input components

### ISSUE THREAD

**Issue by rho66** [primefaces/primefaces#4047]
## 1) Environment
- PrimeFaces version: 6.2
- Does it work on the newest released PrimeFaces version? no
- Does it work on the newest sources in GitHub? no
- Application server + version: wildfly 11.0.0.final
- Affected browsers: Firefox 62.0, Chrome 69.0.3497.92, IE 11

## 2) Expected behavior
I want to highlight required input fields using a css pseudo selecor, depending whether aria-required="true". This works for some input components but not for select input components like selectBooleanButton, selectBooleanCheckbox, triStateCheckbox, selectOneMenu, ...

## 3) Actual behavior
For all input components arial-required should be generated accordingly

## 5) Sample XHTML
[required2.zip](https://github.com/primefaces/primefaces/files/2387409/required2.zip)

**Comment by rho66** [comment]
I fixed it for me using the attached patch ;-)

[aria_required_patch.zip](https://github.com/primefaces/primefaces/files/2387468/aria_required_patch.zip)

**Comment by Rapster** [comment]
Please provide a PR instead

**Comment by rho66** [comment]
My patch was against tag 6_2, so should I try to apply my patches against master branch (6.3-SNAPSHOT) ?

**Comment by Rapster** [comment]
Yes please

**Comment by rho66** [comment]
OK!

**Comment by melloware** [comment]
@rho66 I think I have much cleaner fix for this if you want me to take a shot at it?

**Comment by rho66** [comment]
@melloware Yes, please! It would be great to have a solution without changing existing primefaces code!

**Comment by rho66** [comment]
Great work, melloware!
But some input components are still not rendered correctly to use css pseudo selector as described in expected behavior!

Find below my patches to render components correctly:

[aria_required.patch.gz](https://github.com/primefaces/primefaces/files/2408056/aria_required.patch.gz)

**Comment by melloware** [comment]
Thanks for testing!  I will review your changes and fix it.

**Comment by rho66** [comment]
A special case seems to be InplaceRenderer, where aria-required is rendered correctly but css style 
` [aria-required="true"] {
            border: darkorange thin solid !important;
        }`is not applied?!

In ChipsRenderer I would apply aria-required to the ul-element instead of each li-element!

**Comment by melloware** [comment]
@rho66 Can you code review and test my latest PR: https://github.com/primefaces/primefaces/pull/4071

The only thing I deviated from your fix was in the ColorPickerRenderer. You were only doing it on the "button" mode color picker but not the Inline color picker?  Was there any reason for that?

**Comment by melloware** [comment]
@rho66 for your other issue of Inplace not rendering the proper border style that sounds like a new issue ticket as its not related to this aria-required fix.

**Comment by tandraschko** [comment]
i removed the inplace stuff for now
and i would also think about to ignore this component as it's not really a good design idea to make such a component required
also the logic, to check if it contains a required component requires a complete traversel of the childs and might not be correct in all cases

**Comment by melloware** [comment]
@tandraschko thanks for reviewing and I agree on Inplace. It doesn't feel right to make it required.

**Comment by rho66** [comment]
The basic idea was to highlight required fields by using a different border color as shown in below images. Inspired from an article on stackoverflow ([https://stackoverflow.com/questions/28333772/how-to-show-a-simple-after-a-component-that-is-required]()) I thought that using aria-required would be the simplest way to achive a solution.
From my perspective using inplace with `required="true"` and highlight displayed value is acceptable, because it makes clear which fields must have a value. The grey asterisk '*' makes required fields not that obvious.

- **required input fields highlighted with border color:**

![screenshot038](https://user-images.githubusercontent.com/25029017/45994315-5eeb3b00-c093-11e8-90e1-5fd38440ba2c.png)

- **required inplace fields highlighted with border color:**

![screenshot039](https://user-images.githubusercontent.com/25029017/45994316-5eeb3b00-c093-11e8-8951-2ba5def23af1.png)
